src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1496,6 +1496,8 @@ static unsigned uct_dc_mlx5_ep_fc_hard_req_progress(void *arg)
         /* Force DCI scheduling, since FC resources may never become available
          * unless we send FC_HARD_REQ packet */
         uct_dc_mlx5_ep_schedule(iface, ep, 1);
+        uct_dc_mlx5_iface_progress_pending(iface,
```

Comment:
Maybe call uct_dc_mlx5_iface_dci_get() before uct_dc_mlx5_ep_schedule()?
Calling progress pending seems overkill. Besides, maybe better do it after kh_foreach_key() loop

Reply:
calling `uct_dc_mlx5_iface_dci_get` prior scheduling is a good catch
done